src/navigators/ProfileStackNavigator.tsx

```diff
@@ -54,24 +56,36 @@ const Stack = createNativeStackNavigator<ProfileStackNavigatorParams>();
  */
 export const ProfileStackNavigator = () => {
   const { signOut } = useContext(AuthContext);
+  const { theme } = useContext(ThemeContext);
 
-  const handleSignOut = async () => {
-    try {
-      await signOut();
-    } catch (error: any) {
-      if (error.statusCodes === statusCodes.SIGN_IN_REQUIRED) {
-        Toast.info({
-          content: `No user signed in. Cannot sign out.`,
-          duration: Toast.SHORT,
-        });
-      } else {
-        Toast.info({
-          content: `Failed to sign user out with error: ${error}`,
-          duration: Toast.SHORT,
-        });
+  const logoutButton = useMemo(() => {
```

Comment:
All I'm doing here is wrapping the existing code in `useMemo` to assume the linter this won't be re-rendered each time to avoid performance nightmares.